<a href="https://colab.research.google.com/github/SantiagoVillalobosB/Machine_Learning_2022/blob/PROYECTOS-1-CORTE/PROYECTO_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#### Se importa las bases de datos de wooldridge 
#### Se crea la clase para luego realizar la creación de cada una de las funciones como: 
    Media 
    Mediana
    Desviacion 
    Varianza
    Coeficientes
#### Se realiza un bucle en el cual itere unicamente en las columnas 'wage' y 'educ' e imprima cada resultado de cada funcion ya antes realizada

In [ ]:
!pip install wooldridge
import wooldridge as wd #se importa base
import statistics as st
import scipy.stats as ss


In [ ]:
class estadistica:
    def __init__(self,lista):
        self.lista=lista 
        return None
    #se crea para calcular la media
    def calculadora_media(self):
        return sum(self.lista)/len(self.lista)

    #se crea la desviacon 
    def calculadora_desviacion (self):
        media=estadistica.calculadora_media(self)
        return (sum([(i -media)**2 for i in self.lista])/(len(self.lista-1)))**(1/2)


    #se crea la varianza
    def calculadora_varianza(self):
        return estadistica.calculadora_desviacion(self)**2

    #se crea los coeficientes
    def calculadora_coeficientes(self):
        return ((estadistica.calculadora_desviacion(self))/(estadistica.calculadora_media(self)))
    
    #se crea la mediana
    def calculadora_mediana(self):
        sorted(self.lista)[len(self.lista)//2] 
        if len(self.lista)%2!=0:
            return sorted(self.lista)[len(self.lista)//2] 
        else:
            return (sorted(self.lista)[len(self.lista)//2]+sorted(self.lista)[len(self.lista)//2-1])/2

In [ ]:
datos=wd.data('wage1')
for i in ['wage', 'educ']:
    data=estadistica(datos[i])
    print (i)
    print(f'la media es: {data.calculadora_media()}')
    print(f'la mediana es: {data.calculadora_mediana()}')
    print(f'la desviacion estandar es: {data.calculadora_desviacion()}')
    print(f'la varianza es: {data.calculadora_varianza()}')
    print(f'los coeficientes de variacion son: {data.calculadora_coeficientes()}')

wage
la media es: 5.896102674787035
la mediana es: 4.650000095367432
la desviacion estandar es: 3.6895738135753655
la varianza es: 13.612954925821066
los coeficientes de variacion son: 0.6257648513063981
educ
la media es: 12.562737642585551
la mediana es: 12.0
la desviacion estandar es: 2.766388999158773
la varianza es: 7.652908094666678
los coeficientes de variacion son: 0.220205903988728


In [ ]:
assert data.calculadora_media()==st.mean(datos['educ']), 'no funciona' 
#assert data.calculadora_desviacion()==st.stdev(datos['educ']), 'no funciona' no corre por decimales
#assert data.calculadora_varianza()==st.variance(datos['educ']), 'no funciona' no corre por decimales
#assert data.calculadora_coeficientes()==ss.variation(datos['educ']), 'no funciona' no corre por decimales
assert data.calculadora_mediana()==st.median(datos['educ']), 'no funciona' 

Para hacer el analisis correspondiente sobre los datos obtenidos son:

# ***wage***

Hace referencia al salario promedio que tienen las personas por hora

**Media:** El salario promedio por hora de una persona es aproximadamente de 5.89 dolares la hora, *(se utiliza la medida de dolares debido a que la mayoria de bases de datos de wooldridge son de Estados Unidos)*

**Mediana:** La mediana es de 4.65, es decir que la mitad de la poblacion utilizada como muestra en esta base de datos tiene un salario promedio por hora por encima de 4.65 dolares, y la otra mitad de la muestra tiene un salario promedio por hora inferior a 4.65 dolares

**Desviación estandar:** La desviación estandar nos hace referencia a cuanto estan dispersos los datos, entre más cercano se encuentre a cero este valor menos dispersos se encuentran los datos, es por esta razón que para este dato, la desviación estandar de los salarios es de 3.68 dolares, es un número realmente cercano a cero, entonces quiere decir que los datos están más agrupados hacia la media y no existe tanta dsipersión en los datos.

**Varianza:** La varianza es un estadistico similar a la desviación que busca medir la dispersión de los datos, en este caso para los salarios hay una varianza de 13.61

**Coeficientes de variación:** Como los dos estadisticos anteriores, buscan estimar o conocer la dispersión respecto a la media entre los datos, para el caso del salario el los coeficientes es de un 62%

#***Educ***

Esta variable hace referencia a los años de educación de las personas objeto de estudio de esta base de datos.

**Media:** Los años promedio de educación de una persona es de 12.5 años, es decir que las personas estudiadas tienen doce años y medio como promedio de eduación, se puede decir que aproximadamente la poblacion objetivo tiene minimo el bachillerato completo. 

**Mediana:** La mediana es de 12, es decir que la mitad de la poblacion utilizada como muestra en esta base de datos tiene mas de doce años de educación y la otra mitad de la muestra tiene menos de doce años de escolaridad.

**Desviación estandar:** La desviación estandar para el caso de la educación es de 2.76 años de escolaridad, es un dato muy cercano a cero, y esto da la tranquilidad de que de que no hay tanta dispersion en los datos.

**Varianza:** La varianza para el caso de la educación es de 7.66, es decir que hay una dispersion baja en el caso de la varianza.

**Coeficientes de variación:** Para la variable de educación tienen un coeficiente de variacion del 22%, es decir, que hay una dispersión de los datos con respecto de la media del 22%. 